src/branchless_power_iterator.cpp

```diff
@@ -534,18 +534,12 @@ void BranchlessPowerIterator::normalize_weights(
   double W_pos = 0.;
   Nnet = 0;
   Ntot = 0;
-  Npos = 0;
-  Nneg = 0;
-  for (size_t i = 0; i < next_gen.size(); i++) {
-    if (next_gen[i].wgt > 0.) {
-      W_pos += next_gen[i].wgt;
-      Npos++;
-    } else {
-      W_neg -= next_gen[i].wgt;
-      Nneg++;
-    }
-  }
+  std::tuple<double,double,int,int> tup = kahanBabushkaTuple(next_gen.begin(),next_gen.end());
 
+  W_pos = get<0>(tup);
```

Comment:
There is a really cool c++ feature called `std::tie` that you can use to unpack a tuple. To assign all of these values at once, you can do `std::tie(W_pos, W_neg, Npos, Nneg) = kahanBabushkaTuple(...);`.